File: src/utils/random-domain.ts
import { readCsv } from "./read-csv";

export const randomDomain = (): RandomDomain => {
    const uniNicknames = readCsv('./src/data/uni-nickname.csv');
    const thaiNicknames = readCsv('./src/data/thai-nickname.csv');
    
    const randomUniNickname = uniNicknames[Math.floor(Math.random() * uniNicknames.length)];
    const randomThaiNickname = thaiNicknames[Math.floor(Math.random() * thaiNicknames.length)];
    
    if (!randomUniNickname || !randomThaiNickname || 
        !randomUniNickname.nickname || !randomThaiNickname.en) {
        throw new Error('Failed to get random nicknames');
    }
    
    const uniNicknameVariations = randomUniNickname.nickname;
    if (uniNicknameVariations.length === 0) {
        throw new Error('No nickname variations found');
    }
    
    const randomUniVariation = uniNicknameVariations[Math.floor(Math.random() * uniNicknameVariations.length)];
    if (!randomUniVariation) {
        throw new Error('Failed to get random nickname variation');
    }
    
    const thaiNickname = randomThaiNickname.en[0];
    if (!thaiNickname) {
        throw new Error('Failed to get Thai nickname');
    }
    
    return {
        domain: `${randomUniVariation.toLowerCase()}${thaiNickname.toLowerCase()}@gmail.com`,
        fullName: `${thaiNickname} ${randomUniVariation}`
    }
};

File: migration/src/utils/random_domain.py
import csv
import random
import os

def random_domain():
    current_dir = os.path.dirname(os.path.abspath(__file__))
    data_dir = os.path.join(current_dir, '..', 'data')
    thai_nickname_file = os.path.join(data_dir, 'thai-nickname.csv')
    uni_nickname_file = os.path.join(data_dir, 'uni-nickname.csv')
    
    with open(thai_nickname_file, 'r', encoding='utf-8') as file:
        reader = csv.DictReader(file)
        thai_nicknames = list(reader)
    
    with open(uni_nickname_file, 'r', encoding='utf-8') as file:
        reader = csv.DictReader(file)
        uni_nicknames = list(reader)
    
    thai_nickname = random.choice(thai_nicknames)
    uni_nickname = random.choice(uni_nicknames)
    
    return {
        'fullName': f"{thai_nickname['en'].lower().capitalize()} {uni_nickname['firstname'].lower().capitalize()}",
        'domain': f"{thai_nickname['en'].lower()}{uni_nickname['firstname'].lower()}{uni_nickname['nickname'].lower()}"
    } 